src/HIE/Bios/Cradle.hs

```diff
@@ -796,6 +815,23 @@ cabalGhcDirs l cabalProject workDir = do
   where
     projectFileArgs = projectFileProcessArgs cabalProject
 
+cabalPathCompilerPath :: LogAction IO (WithSeverity Log) -> FilePath -> CradleProjectConfig -> CradleLoadResultT IO (Maybe FilePath)
+cabalPathCompilerPath l workDir projectFile = do
+  liftIO (getCabalVersion l workDir) >>= \case
+    Just cabal_version | cabal_version >= makeVersion [3,14] -> do
```

Comment:
I think it makes sense to extract this check into its own function with a name and documentation, similar to `isCabalMultipleCompSupported`.

Reply:
Wanted to make it consistent with `isCabalMultipleCompSupported` and just realised `ProgramVersions` was actually available on both codepaths that reach here. This cut redundant invocations to `cabal --version`, at the tune of 50ms each on my machine.